### PR TITLE
Milestone-5/Blog app - views

### PR DESCRIPTION
**Pull request by edwardoluyara**
Here in this milestone I have been able to implement the following:
- Implement the design from the sneak peek wireframes.
  - Styling is nice to have but focus on making sure everything functions first.
- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -53,9 +63,14 @@
...
+  add_foreign_key "comments", "posts"
   add_foreign_key "comments", "posts", column: "posts_id"
+  add_foreign_key "comments", "users", column: "author_id"
   add_foreign_key "comments", "users", column: "users_id"
+  add_foreign_key "likes", "posts"
   add_foreign_key "likes", "posts", column: "posts_id"
+  add_foreign_key "likes", "users", column: "author_id"
   add_foreign_key "likes", "users", column: "users_id"
+  add_foreign_key "posts", "users", column: "author_id"
   add_foreign_key "posts", "users", column: "users_id"
```
Comment (Deepakdanger) [inline db/schema.rb]: - [ ] Here you have added column as `author_id` but you have not removed `user_id`. This is creating issue while creating posts, comments, or likes. Please fix this. First you have to rollback and then migrate the right command.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+  </section>
+
+  <section>
+    <h2> Bio </h2>
+
+    <p> <%= @user.bio %> </p>
+  </section>
+
+  <section>
+    <% @user.three_most_recent_posts.each do |post| %>
```
Comment (Deepakdanger) [inline app/views/users/show.html.erb:24]: - [ ] Also getting an error on the user's show page. Please fix this.

Screenshot from 2022-05-04 03-02-49

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @oluyaratosin123, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @oluyaratosin123 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 

## Optional suggestions

image

- [ ] Kindly make sure you fix this issue when a comment is created(names in convention). This accused because your association contains an error, the association `belong_to` always refers to the table but in the singular form(`belong_to: post` not `belong_to: posts` ).  
#### In your models/comment.rb 
https://github.com/oluyaratosin123/Blog_App_With_Rails/blob/b4041c6224c3432a88c87c63cb11d434f7322cf1/app/models/comment.rb#L3

#### User this :point_down: 
```rb
class Comment < ApplicationRecord
  belongs_to :author, class_name: 'User'
  belongs_to :post

  after_save :update_comments_counter

  private

  def update_comments_counter
    post.increment!(:comments_counter)
  end
end
```
#### Instead of this :point_down: 

```rb
class Comment < ApplicationRecord
  belongs_to :author, class_name: 'User'
  belongs_to :posts

  after_save :update_comments_counter

  private

  def update_comments_counter
    posts.increment!(:comments_counter)
  end
end
```

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**